currency-rates/internal/handlers/subscribe.go

```diff
@@ -56,5 +59,32 @@ func (handler SubscribeHandler) CreateSubscription(w http.ResponseWriter, r *htt
 
 func isValidEmail(email string) bool {
 	return emailRegex.MatchString(email)
-  }
-  
\ No newline at end of file
+}
+  
+func (handler SubscribeHandler) DeactivateSubscription(w http.ResponseWriter, r *http.Request) {
+	err := r.ParseForm()
+	if err != nil {
+		http.Error(w, "Form parse error", http.StatusBadRequest) 
+		return
+	}
+
+	formData := r.Form
+
+	email := formData.Get("email")
+	
+	err = handler.repo.DeactivateEmail(r.Context(), email)
+	responseStatus := http.StatusOK
+	
+	if err != nil {
+		if errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
This is infrastructure specific errors. What if you will change orm from gorm to somehting another, so you will need to fix this part of code, that shouldn't depend on orm that you are using.

The best case will be to check this error in Repo infrastructure and send domain error like EntityNotFound. So here will be constant check, not depending on orm, but on domain logic.

Reply:
done